core/src/test/java/org/apache/calcite/util/format/FormatElementEnumTest.java

```diff
@@ -33,4 +33,130 @@ class FormatElementEnumTest {
     FormatElementEnum.DAY.format(ts, Date.from(Instant.parse("2014-09-30T10:00:00Z")));
     assertThat(ts, hasToString("Tuesday"));
   }
+
+  @Test void testD() {
+    StringBuilder ts = new StringBuilder();
+    FormatElementEnum.D.format(ts, Date.from(Instant.parse("2014-09-30T10:00:00Z")));
+    assertThat(ts, hasToString("3"));
+  }
+
+  @Test void testDD() {
+    StringBuilder ts = new StringBuilder();
+    FormatElementEnum.DD.format(ts, Date.from(Instant.parse("2014-09-30T10:00:00Z")));
+    assertThat(ts, hasToString("30"));
+  }
+
+  @Test void testDDD() {
+    StringBuilder ts = new StringBuilder();
+    FormatElementEnum.DDD.format(ts, Date.from(Instant.parse("2014-09-30T10:00:00Z")));
+    assertThat(ts, hasToString("273"));
+  }
+
+  @Test void testDY() {
+    StringBuilder ts = new StringBuilder();
+    FormatElementEnum.DY.format(ts, Date.from(Instant.parse("2014-09-30T10:00:00Z")));
+    assertThat(ts, hasToString("Tue"));
+  }
+
+  @Test void testFF1() {
+    StringBuilder ts = new StringBuilder();
+    FormatElementEnum.FF1.format(ts, Date.from(Instant.parse("2014-09-30T10:00:00Z")));
+    assertThat(ts, hasToString("0"));
+  }
+
+  @Test void testFF2() {
```

Comment:
if these are microseconds perhaps you can use a datetime with microseconds?
Does Java support microseconds in Date?

Reply:
I didn't find the corresponding API

Reply:
What API didn't you find?
It looks like Instant.parse supports microseconds and date supports milliseconds.

Reply:
Sorry, I misunderstood the meaning before, but I have corrected it now